src/Controller/SecurityController.php

```diff
@@ -33,4 +42,49 @@ public function logout()
     {
         throw new \LogicException('This method can be blank - it will be intercepted by the logout key on your firewall.');
     }
+
+    /**
+     * @Route("/admin/user/{id}/set-password", name="admin_user_set_password")
+     * @IsGranted("ROLE_ADMIN")
+     */
+    public function setUserPassword(Request $request, User $user, TranslatorInterface $translator, UserManager $userManager, EntityManagerInterface $entityManager)
+    {
+        $minLength = SetUserPasswordFormType::MIN_LENGTH;
+
+        $form = $this->createForm(SetUserPasswordFormType::class, [
+            'password' => ByteString::fromRandom(2 * $minLength),
+        ]);
+
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $password = trim($form->getData()['password'] ?? '');
+            if (strlen($password) < $minLength) {
+                $this->addFlash('danger', $translator->trans('User password too short. Must be contain least %min_length% characters.', [
+                    '%min_length%' => $minLength,
+                ]));
+            } else {
+                try {
+                    $userManager->setPassword($user, $password);
+                    $userManager->updateUser($user);
+
+                    $this->addFlash('success', $translator->trans('Password for user %username% set to %password%', [
+                        '%username%' => $user->getUsername(),
+                        '%password%' => $password,
+                    ]));
+
+                    return $this->redirectToRoute('easyadmin', ['entity' => 'User']);
+                } catch (\Throwable $t) {
+                    $this->addFlash('danger', $translator->trans('Error settings password for user %username%: %message%', [
```

Comment:
Should probably just say `setting` and is also missing a translation.

Reply:
Yes,